adlfs/core.py

```diff
@@ -743,6 +743,9 @@ def _put_data(rest, op, path, data, max_attempts=10, **kwargs):
         except (PermissionError, FileNotFoundError):
             raise
         except Exception as e:
+            if '"BadOffsetException"' in repr(e):
```

Comment:
@martindurant is it possible for this check to be more targeted to just the uploader logic? Callers should be able to decide how they want to handle badOffsetExceptions when they build their own apps that use the SDK, we don't want to swallow this exception for all cases, just in our implementation of the uploader where we know with near certainty what is going on.